Update CHANGELOG to document 7.2 items (#4497)
ROCm/AMDMIGraphX: CHANGELOG.md

Full documentation for MIGraphX is available at
[https://rocmdocs.amd.com/projects/AMDMIGraphX/en/latest/](https://rocmdocs.amd.com/projects/AMDMIGraphX/en/latest/).

## MIGraphX 2.15 for ROCm 7.2.0

### Added

* Added MXFP4 support for Quark and Brevitas quantized models

> * Added a mxr file in /tmp on compile error

Author: Moved to Changed section and modified text to ...
Model compilation failures now produce an mxr file for debugging the failure

* Added dynamic shape support for DepthToSpace Op

> * Added updated inputs for multihead attention

Reviewer: Can this be a bit more precise? Or will users of multihead attention now what this is?

* Added GEMM+GEMM fusions

> * Added dim_params input to parse_onnx python api

Reviewer: ```suggestion
* Added the `dim_params` input parameter to the `parse_onnx` python call.
```

* Created an API to query supported ONNX Operators
* Added right pad masking mode for Multihead Attention
* Added support for Flash Decoding
* Added Torch-MIGraphX installation instructions

> * Added Op Builders and documentation

Reviewer: ```suggestion
* Added Operator Builders with supporting documentation
```

### Changed

> * Updated Resize op to support linear mode

Reviewer: ```suggestion
* Updated the Resize operator to support linear mode
```

* Add index range check in Gather Operator
* Use --input-dim for specifying dynamic shapes at driver runtime

> * Dont fuse pointwise when its used across submodules

Author: Reading the Motivation behind the PR I think this is actually a bug fix.  We hit a model that never completed the compile.  It would sit forever in a hang.  I will change the section and text...

 * Resolved a compilation hang issue (#4428)

* Support different stride in if branches with ONNX
* ONNX version change to 1.18.0 to support PyTorch 2.9
* Refactor GroupQueryAttention
* Enable PipelineRepoRef parameter in CI
* Hide LLVM symbols that come from ROCmlir and provide option for stripping in release mode

> ### Fixed

Reviewer: ```suggestion
### Resolved issues
```

* Fixed `pointwise: Wrong number of arguments` error when quantizing certain models to `int8` [(#4398)](https://github.com/ROCm/AMDMIGraphX/pull/4398).
* TopK exception bugfix (#4329)
* Updated SD3 example for change in optimum-onnx[onnxruntime] (#4344)

> * Check if instruction is actually a literal (#4388)

Reviewer: This is very vague. Can context be added? For example, in which function/operator/other was this happening?

Author: Motivation
The compare_literals function was calling instruction::get_literal() on instructions that were not necessarily literal instructions. The function assumed that if an instruction could be evaluated (eval() returns non-empty), it was safe to call get_literal(), but this assumption was incorrect.

Rewording to...
* Fixed an issue with Torch-MIGraphX where the model compilation would fail (#4388)

> * Fix wrong number of arguments for pointwise (#4398)

Author: Yes the PR numbers are duplicates.  I'll remove the line

* Fix issue when broadcasting a reduction with different dimensions as the input (#4408)
* Fixing invalid characters inside path name on Windows (#4420)
* Quiet nrvo and noreturn warnings (#4429)
* Fix "reduce_sum: axes: value out of range" error in simplify_reshapes (#4443)
* Updated README `rbuild` installation instructions to use python venv to avoid warning (#4405)
* Create directory before dumping (#4383)

### Optimized

> * Bump SQlite3 to 3.50.4

Reviewer: This should be under Changed

> * Lower lrn to pooling

Author: * Converted the `LRN` operator to an optimized `pooling` operator

> * Streamline find_matches function

Reviewer: ```suggestion
* Streamlined the `find_matches` function
```
